src/components/drawing/index.js

```diff
@@ -271,24 +272,83 @@ function singlePointStyle(d, sel, trace, markerScale, lineScale, marker, markerL
             });
     }
     else {
-        sel.style('stroke-width', lineWidth + 'px')
-            .call(Color.fill, fillColor);
+        sel.style('stroke-width', lineWidth + 'px');
+
+        var markerGradient = marker.gradient;
+        var gradientColor = markerGradient && (d.mgc || markerGradient.color);
+        var gradientType = d.mgt || markerGradient.type;
+        if(gradientType && gradientType !== 'none') {
+            sel.call(drawing.gradient, gd, gradientType, fillColor, gradientColor);
+        }
+        else {
+            sel.call(Color.fill, fillColor);
+        }
+
         if(lineWidth) {
             sel.call(Color.stroke, lineColor);
         }
     }
 }
 
-drawing.singlePointStyle = function(d, sel, trace) {
-    var marker = trace.marker,
-        markerLine = marker.line;
+var HORZGRADIENT = {x1: 1, x2: 0, y1: 0, y2: 0};
+var VERTGRADIENT = {x1: 0, x2: 0, y1: 1, y2: 0};
 
-    // allow array marker and marker line colors to be
-    // scaled by given max and min to colorscales
-    var markerScale = drawing.tryColorscale(marker, ''),
-        lineScale = drawing.tryColorscale(marker, 'line');
+drawing.gradient = function(sel, gd, type, color1, color2) {
```

Comment:
I was debating whether this goes in `color` or `drawing` - ended up deciding on `drawing` primarily because it's tied to `gd` (so it can create `defs` in the right place, without having to walk the tree with `Lib.getPlotDiv` for every marker) and nothing else in `color` is... but could be persuaded otherwise.